### PR TITLE
Add more scalastyle rules

### PR DESCRIPTION
**Pull request by ceedubs**
The intention of this is so that potential contributors see a consistent
style and so that they can check the scalastyle output and hopefully
avoid reviewers leaving nitpicky formatting comments on PRs.

I'd like to wait until #845 is merged before we merge this one so we don't potentially cause yet another merge conflict in that PR. I wanted to go ahead and submit this so that I don't forget to though.

### CODE REVIEW

**Comment by peterneyens** [comment]
Maybe we could also add a rule to check we write type class instead of typeclass, so we don't need to clean this up regularly like in  #1149.

**Comment by ceedubs** [comment]
@peterneyens that is an excellent idea. I'm very inconsistent on this point.

I think that there are a couple difficulties though.
- I don't think scalastyle runs on our docs, so it's still up to us to try to be consistent there.
- We'll probably have to make the pattern not match if there's a leading `@` to avoid our simulacrum annotations tripping this.

**Comment by ceedubs** [comment]
@peterneyens updated to include "type class" rule.

**Comment by peterneyens** [comment]
Maybe we could also remove the `NumberOfTypes` rule in this PR, like we mentioned in #1093 ?

**Comment by ceedubs** [comment]
#845 has been merged and I've updated this PR. I think it's ready for review/merge.

**Comment by kailuowang** [comment]
👍  LGTM

File: core/src/main/scala/cats/data/OneAnd.scala [hunk 1]
```diff
@@ -27,7 +27,7 @@ final case class OneAnd[F[_], A](head: A, tail: F[A]) {
    */
   def filter(f: A => Boolean)(implicit F: MonadCombine[F]): F[A] = {
     val rest = F.filter(tail)(f)
-    if (f(head)) F.combineK(F.pure(head), rest) else rest
+    if(f(head)) F.combineK(F.pure(head), rest) else rest
```
Comment (fthomas) [inline core/src/main/scala/cats/data/OneAnd.scala]: Later in this diff, spaces are inserted between `if` and `(`, for example [here](https://github.com/typelevel/cats/pull/1142/files#diff-26729163874fb5967d1a0c27b6db2457R60). Can we add a scalastyle rule to make this consistent?

**Comment by fthomas** [comment]
I made a minor comment about spaces after `if`. Apart from that, this LGTM. :+1:

File: core/src/main/scala/cats/data/OneAnd.scala (hunk 1 above)
Comment (ceedubs) [inline core/src/main/scala/cats/data/OneAnd.scala]: Wow good catch. The other places were caught by `EnsureSingleSpaceAfterTokenChecker`. It seems to me like a scalastyle bug that this particular place isn't caught by that rule.

**Comment by ceedubs** [comment]
@fthomas I've added a rule for space after `if` since for some reason `EnsureSingleSpaceAfterTokenChecker` wasn't working in certain cases.

**Pull request merged**